src/coreclr/gc/gcpriv.h

```diff
@@ -3382,13 +3391,23 @@ class gc_heap
 #endif //BACKGROUND_GC
 
 #ifdef USE_REGIONS
-// This is the number of regions we would free up if we sweep.
-// It's used in the decision for compaction so we calculate it in plan.
+    // This is the number of regions we would free up if we sweep.
+    // It's used in the decision for compaction so we calculate it in plan.
     PER_HEAP_FIELD_SINGLE_GC int num_regions_freed_in_sweep;
 
     PER_HEAP_FIELD_SINGLE_GC int sip_maxgen_regions_per_gen[max_generation + 1];
     PER_HEAP_FIELD_SINGLE_GC heap_segment* reserved_free_regions_sip[max_generation];
 
+    PER_HEAP_FIELD_SINGLE_GC int regions_per_gen[max_generation + 1];
+
+    // Used to keep track of how many regions we have planned to see if any generation
+    // doens't have a region yet and act accordingly.
+    PER_HEAP_FIELD_SINGLE_GC int planned_regions_per_gen[max_generation + 1];
+
+    PER_HEAP_FIELD_SINGLE_GC int new_gen0_regions_in_plns;
+    PER_HEAP_FIELD_SINGLE_GC int new_regions_in_prr;
+    PER_HEAP_FIELD_SINGLE_GC int new_regions_in_threading;
```

Comment:
These three fields seem to be used for diagnostic purposes only, correct?

Reply:
fixed